fix: junit testresult file generation

jverelst opened the pull request:
Dear maintainers,

This is the first PR I ever created for the SFDX repositories, I hope I properly followed the contributing guidelines and code styles. Please let me know if there is anything that I should change in the PR.

### What does this PR do?

Resolved an issue where the junit file generation created invalid XML files, because the failure message
contains characters that should be XML-escaped. These are: <, >, &, " and '.
Made sure to xml-encode these in the message, updated a testcase to make sure that this is now covered in
a test as well.

### What issues does this PR fix or reference?

This PR fixes the following issue I logged: forcedotcom/salesforcedx-apex#280

### Functionality Before

The junit generated created invalid XML, because XML characters like the quote were not properly escaped.

### Functionality After

Valid XML generation

jverelst commented:
I have added the mail address to my github account and have just signed the CLA using the same mailaddress.

gbockus-sf commented:
Hi @jverelst
Thanks so much for your contribution!  Could you run the `yarn lint` script and address the few issues that are being reported? It looks like just a few swaps to using const. Thanks again!

gbockus-sf commented on packages/apex-node/src/reporters/junitReporter.ts:
```diff
@@ -72,8 +72,9 @@ export class JUnitReporter {
     let junitTests = '';
 
     for (const testCase of tests) {
+      let methodName = this.xmlEscape(testCase.methodName);
```
use const to make the linter happy

gbockus-sf commented on packages/apex-node/src/reporters/junitReporter.ts:
```diff
@@ -82,7 +83,8 @@ export class JUnitReporter {
         testCase.outcome === ApexTestResultOutcome.CompileFail
       ) {
-        junitTests += `${tab}${tab}${tab}<failure message="${testCase.message}">`;
+        let message = this.xmlEscape(testCase.message);
```
use const.

gbockus-sf reviewed (changes requested):
Just a couple of lint fixes.

jverelst commented:
@gbockus-sf thanks for the review and feedback, I have updated my PR accordingly.

I've added 2 commits; first one with the let/const change, second one with some whitespace changes to make the linter happy. Hope everything is in order now!

randi274 reviewed (changes requested):
Hey @jverelst - I was just digging into some weird build errors with this PR, but noticed that you added tests to the tapReporter, not the junitReporter. Could you also update the junitReporter test too since that's where you made the changes? We can keep the tapReporter change too as an extra sanity check 👍

arturbianchi commented:
Hi guys. Any news for this bug? I'm facing the same issue.

alexmbaker commented:
@randi274  responding to your observation about any new junitReporter tests for this issue.

While there have been no changes to the code in junitReporter.tests.ts the test scenarios defined in testResults.ts have been updated so that the source data includes characters that need escaping in an XML document.  The expected output for the junit report has been updated to reflect the updated test scenario and now includes some XML escape character sequences.

The update to the tapReporter test is to align the expected result in with the updated test scenario.   

It would be great if we can progress this PR, I am happy to contribute if there is anything else needed to proceed?

randi274 commented:
Thanks for your patience on this @jverelst! We had some PTO, build issues, and a small fix to make, but this has been merged in with #285. It should be available likely next week with the CLI upgrade. Thanks for your contribution as well!
